deploy/kubernetes/configmap.yaml

```diff
@@ -6,31 +6,79 @@ metadata:
 data:
   ncp.ini: |
 
-    [vc]
 
-    # IpAddress or Hostname of VC
-    #vc_endpoint = <None>
+    [DEFAULT]
+
+    # If set to true, the logging level will be set to DEBUG instead of the
+    # default INFO level.
+    #debug = False
+
+    # If set to true, log output to standard error.
+    #use_stderr = True
+
+    # Destination to send api log to. STDOUT or STDERR for console output. FILE
+    # to write log to file configured in "api_log_file". NONE to disable api
+    # log.
+    # Choices: STDOUT STDERR FILE NONE
+    #api_log_output = NONE
+
+    # Name of log file to send API access log to.
+    #api_log_file = ncp_api_log.txt
+
+    # Interval in seconds to logs api call to output configured in
+    # api_log_output
+    #api_log_interval = 60
+
+    # When api_log_output is not NONE, this option determines if api calls
+    # should be collected per NSX cluster or individual NSX manager.
+    # Choices: API_LOG_PER_ENDPOINT API_LOG_PER_CLUSTER
+    #api_log_mode = API_LOG_PER_ENDPOINT
+
+    # If set to true, use syslog for logging.
+    #use_syslog = False
+
+    # The base directory used for relative log_file paths.
+    #log_dir = <None>
+
+    # Name of log file to send logging output to.
+    #log_file = <None>
+
+    # max MB for each compressed file. Defaults to 100 MB.
+    #log_rotation_file_max_mb = 100
+
+    # max MB for each compressed file for API logs.Defaults to 10 MB.
+    #api_log_rotation_file_max_mb = 10
+
+    # Total number of compressed backup files to store. Defaults to 5.
+    #log_rotation_backup_count = 5
+
+    # Total number of compressed backup files to store API logs. Defaults to 5.
+    #api_log_rotation_backup_count = 5
+
+    # Log level for the root logger. If debug=True, the default root logger
+    # level will be DEBUG regardless of the value of this option. If this
+    # option is unset, the default root logger level will be either DEBUG or
+    # INFO according to the debug option value
+    # Choices: NOTSET DEBUG INFO WARNING ERROR CRITICAL
+    #loglevel = <None>
 
-    # The SSO domain associated with the deployment
-    #sso_domain = vsphere.local
 
-    # VC API server HTTPS port.
-    #https_port = 443
 
 
     [coe]
 
     # Container orchestrator adaptor to plug in.
-    adaptor = kubernetes
+    #adaptor = kubernetes
```

Comment:
Why comment `adaptor`

Reply:
Because it's automatically generated from the jinja2 template in NCP. i.e. in the other all-in-one yaml, there's also `#adaptor = kubernetes`. And in NCP config we had defined the default value to "kubernetes" so it won't introduce issue. I don't want to uncomment it because if we do it we should uncomment it every time when updating the file, unless we change the jinja2 template in NCP, which change also will impact the content of other all-in-one yamls. (That's also why I reserved some ugly multiple-line spaces because I think we'd better not prune it by manual, for the complete automation.)

Reply:
I just saw it's different with the options in openshift. I'm fine with current automatic way.

Reply:
May I ask that the value of entry commented out is the same with NCP default config value?

Reply:
> May I ask that the value of entry commented out is the same with NCP default config value?

yes

Reply:
I recovered to `adaptor = kubernetes` because the operator reported the error "failed to get key adaptor from section coe" in Kubernetes platform. This is because that we don't have `FillDefaults` for Kubernetes platform and the validation will check if the value exists. Some alternative solutions:
1. Add `FillDefaults` function for Kubernetes
2. Remove the valudation function for Kubernetes
3. Modify the render script in NCP to uncomment the value to make sure the generated yaml doesn't comment the value
I'll try 3 after the patch was merged.